frontend/src/app/(navfooter)/new/NewScratchForm.tsx

```diff
@@ -257,6 +262,34 @@ export default function NewScratchForm({
         }
     };
 
+    useEffect(() => {
+        if (!label) {
+            // reset potential duplicates if no diff label
+            setDuplicates([]);
+            return;
+        }
+
+        const filterCandidates = (scratches: TerseScratch[]) => {
+            return scratches.filter((scratch: TerseScratch) => {
+                // search endpoint is greedy, so only match whole-name
+                if (scratch.name !== label) {
+                    return false;
+                }
+                // filter on preset if we have it
+                if (typeof presetId !== "undefined") {
+                    return scratch.preset === presetId;
+                }
+                // otherwise filter on platform
+                return scratch.platform === platform;
+            });
+        };
+
+        get(`/scratch?search=${label}`)
+            .then((x) => x.results)
+            .then(filterCandidates)
+            .then(setDuplicates);
+    }, [label, platform, presetId]);
```

Comment:
Since label is a text input, every character the user types into it will trigger a network request, which we don't want. I'd probably wrap use `useDebounce`, like this code that will only make the api call 1000ms after the user finishes typing: https://github.com/decompme/decomp.me/blob/ae0529095372de677e9f6660aab8328dd55c5f95/frontend/src/components/Scratch/panels/DecompilePanel.tsx#L24-L39